custom_components/spotcast/__init__.py

```diff
@@ -214,15 +228,34 @@ def start_casting(call: ha_core.ServiceCall):
             uri = uri.split(":")
             uri[0] = uri[0].lower()
             uri[1] = uri[1].lower()
-            uri = ':'.join(uri)
+            uri = ":".join(uri)
 
         # first, rely on spotify id given in config otherwise get one
         if not spotify_device_id:
             spotify_device_id = spotcast_controller.get_spotify_device_id(
                 account, spotify_device_id, device_name, entity_id
             )
 
-        if is_empty_str(uri) and is_empty_str(search) and is_empty_str(category):
+        if (
+            is_empty_str(uri)
+            and len(
+                list(
+                    filter(
+                        lambda x: not is_empty_str(x),
+                        [
+                            artistName,
+                            playlistName,
+                            trackName,
+                            showName,
+                            episodeName,
+                            audiobookName,
+                            genreName,
+                        ],
+                    )
+                )
+            )
+            == 0
+        ):
```

Comment:
I haven't tested this. The intention is, transfer playback only in the case where none of the search fields are set.